### ISSUE TITLE
Abstract out configuration logic

### ISSUE THREAD

**Issue by hinerm** [imagej/napari-imagej#208]
Looking at [this code](https://github.com/imagej/napari-imagej/blob/4b8646ad2a14f6b6476b87ac05b9bd56e48dfb96/src/napari_imagej/__init__.py#L38-L43) it feels odd to have consideration for a test mode in the main code.

Is the napari-imagej configuration *always* being read from a local user directory? Including when running tests?

Ideally the path should be something like:
```
<source of configuration> --> configuration object --> napari-imagej
```

Where the source is a configuration file via `confuse` when running via the app, but an alternative is used for testing (either hard-coded or via a test-specific configuration file)

**Comment by ctrueden** [comment]
Thanks @hinerm, I was just writing up some of my thoughts in this area as well:

We should do some refactoring to better separate the napari-imagej business logic from the user interface. For example:

* PyImageJ initialization logic should be pulled out of the asynchronous GUI initialization layer, such that napari-imagej can be initialized via the same code path when unit testing, as happens asynchronously in the napari widget. In particular, the `java.py` should not import QT threading classes, but rather the async logic requiring those classes should go into a separate file, probably in `widgets`.
* All GUI commands, e.g. sending data napari &lrarr; ImageJ2, should happen in dedicated functions that are independent of GUI, and GUI element callback functions e.g. button presses should do nothing except call those same functions. It should never be necessary for business logic to programmatically trigger UI events e.g. button presses.
* The napari-imagej settings should not be confuse-specific, but rather act as a simple dictionary, which *may* be loaded via confuse and/or config files, but also may be specified directly for use e.g. in unit tests. No unit test should hack the user's config file, and the `NAPARI_IMAGEJ_TESTING` environment variable case logic should not be needed.

I hadn't finished substantiating all of those points with code examples, so I'm not 100% confident some of these points are really significant issues, but I want to make sure our architecture is rock solid before we commit to a 1.0.0 release!

**Comment by ctrueden** [comment]
Resolved by #226.
